crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -138,16 +134,25 @@ def bag_item
 
     private
 
+    def repos_dir
+      File.join("/srv/tftpboot", @platform, "repos")
+    end
+
+    def repodata_path
+      File.join(repos_dir, @config["name"], "repodata")
+    end
+
     #
     # validation helpers
     #
-    def all_repo_dirs
-      path = File.join("/srv/tftpboot", @platform, "repos")
-      Dir["#{path}/*"].map { |p| p.split("/").last }
+    def check_directory
+      Dir.exist? File.join(repos_dir, @config["name"])
     end
 
     def check_repo_tag
-      expected = @registry["repomd"]["tag"]
+      expected = @config.fetch("repomd", {})["tag"]
```

Comment:
Why? This just ensures that we can get a value (even nil) if there's no repomd attribute.